clang/include/clang/Basic/Attr.td

```diff
@@ -1431,15 +1432,32 @@ def LoopUnrollHint : StmtAttr {
 }
 
 def IntelReqdSubGroupSize: InheritableAttr {
-  let Spellings = [GNU<"intel_reqd_sub_group_size">,
-                   CXX11<"intel", "reqd_sub_group_size">];
+  let Spellings = [
+    GNU<"intel_reqd_sub_group_size">, CXX11<"intel", "reqd_sub_group_size">,
+    CXX11<"intel", "sub_group_size"> // SYCL2020 spelling.
+  ];
   let Args = [ExprArgument<"Value">];
   let Subjects = SubjectList<[Function], ErrorDiag>;
   let Documentation = [IntelReqdSubGroupSizeDocs];
   let LangOpts = [OpenCL, SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
   let SupportsNonconformingLambdaSyntax = 1;
+  let Accessors =
+    [Accessor<"isSYCL2020Spelling", [CXX11<"intel", "sub_group_size">]>];
 }
 
+def IntelNamedSubGroupSize : InheritableAttr {
+  let Spellings = [CXX11<"intel", "named_sub_group_size">];
+  let Args = [EnumArgument<"Type", "SubGroupSizeType", ["automatic", "primary"],
+                           ["Automatic", "Primary"]>];
+  let Subjects = SubjectList<[Function], ErrorDiag>;
+  let Documentation = [IntelNamedSubGroupSizeDocs];
+  let LangOpts = [OpenCL, SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
+  let SupportsNonconformingLambdaSyntax = 1;
+}
+
+def : MutualExclusions<
+          [IntelReqdSubGroupSize, IntelNamedSubGroupSize, SYCLSimd]>;
```

Comment:
I have no idea why GitHub decided that previous comment was pending despite submitting it in the last review. Neato. Adding this comment to hopefully get my previous comment to appear, but we'll see.

Reply:
Won't this make me fail the clang-format bot though?

Reply:
Er, good question -- does our CI run clang-format over .td files, @bader? That would be a bit of a surprise (I've seen plenty of cases where clang-format starts inserting a ton of whitespace in really awful places in .td files).

Reply:
The check was disabled for almost a day and I recently re-enabled it. It should be ran for next update.

It might be worth mentioning that we use quite old version of clang-format tool. I've opened https://github.com/intel/llvm/pull/3540 to update it.

Reply:
I think it's a mistake to require people to use clang-format in Attr.td because it doesn't always result in particularly good changes. As a trivial example:

What's in Attr.td.
```
def Ownership : InheritableAttr {
  let Spellings = [Clang<"ownership_holds">, Clang<"ownership_returns">,
                   Clang<"ownership_takes">];
  let Accessors = [Accessor<"isHolds", [Clang<"ownership_holds">]>,
                   Accessor<"isReturns", [Clang<"ownership_returns">]>,
                   Accessor<"isTakes", [Clang<"ownership_takes">]>];
  let AdditionalMembers = [{
    enum OwnershipKind { Holds, Returns, Takes };
    OwnershipKind getOwnKind() const {
      return isHolds() ? Holds :
             isTakes() ? Takes :
             Returns;
    }
  }];
  let Args = [IdentifierArgument<"Module">,
              VariadicParamIdxArgument<"Args">];
  let Subjects = SubjectList<[HasFunctionProto]>;
  let Documentation = [Undocumented];
}
```
What clang-format does to this code:
```
def Ownership : InheritableAttr {
  let Spellings = [
    Clang<"ownership_holds">, Clang<"ownership_returns">,
    Clang<"ownership_takes">
  ];
  let Accessors = [
    Accessor<"isHolds", [Clang<"ownership_holds">]>,
    Accessor<"isReturns", [Clang<"ownership_returns">]>,
    Accessor<"isTakes", [Clang<"ownership_takes">]>
  ];
  let AdditionalMembers = [{
    enum OwnershipKind { Holds, Returns, Takes }
    ;
    OwnershipKind getOwnKind() const {
      return isHolds() ? Holds : isTakes() ? Takes : Returns;
    }
  }];
  let Args = [IdentifierArgument<"Module">, VariadicParamIdxArgument<"Args">];
  let Subjects = SubjectList<[HasFunctionProto]>;
  let Documentation = [Undocumented];
}
```
So it's not awful, but it can harm readability to require it. (There are other examples where clang-format gets confused, but the key bit here is that it gets confused about source code within tablegen files.)

Reply:
FWIW: I made your change and the clang-format bot didn't seem to care... so yay ?:)

Reply:
I'm counting it as a yay. :-D